### ISSUE TITLE
[BUG] System Restore Stuck at Pending due to Tolerations not Applied

### ISSUE THREAD

**Issue by Jaryllan** [longhorn/longhorn#8656]
## Describe the bug

When initiated a system restore. A pod job with prefix "longhorn-system-rollout-xxxx" will stuck at pending. Found out that the tolerations that are applied to other longhorn pods is missing for longhorn-system-rollout job. 

![longhorn-restore-pending](https://github.com/longhorn/longhorn/assets/32004583/f641dd7a-a72d-4e28-a8cf-f46a5cc564dc)

## To Reproduce

1. Install longhorn chart with tolerations configured.

```
defaultSettings:
  backupTarget: <hidden>
  backupTargetCredentialSecret: <hidden>
  taintToleration: node-role.kubernetes.io/control-plane=:NoSchedule
longhornDriver:
  nodeSelector:
    node-role.kubernetes.io/control-plane: null
  tolerations:
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
    operator: Exists
longhornManager:
  log:
    format: plain
  nodeSelector:
    node-role.kubernetes.io/control-plane: null
  tolerations:
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
    operator: Exists
longhornUI:
  nodeSelector:
    node-role.kubernetes.io/control-plane: null
  replicas: 2
  tolerations:
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
    operator: Exists
```

2. Making sure taint tolerations is in longhorn portal -> Settings -> General -> Kubernetes taint tolerations
3. Create system backup
4. Detach volumes
5. Create System restore

## Expected behavior

longhorn-system-rollout pod should include with the configured tolerations and run without stuck in pending.

**Comment by innobead** [comment]
cc @c3y1huang

**Comment by derekbit** [comment]
cc @c3y1huang

**Comment by c3y1huang** [comment]
I am able to reproduce the issue following the steps outlined in the issue description.

**Comment by innobead** [comment]
The backport for 1.7 has been merged, so it's for 1.7.2 instead of 1.7.3. cc @derekbit

**Comment by yangchiu** [comment]
Verification passed on master-head (longhorn-manager 4d157f8) following [the test steps](https://github.com/longhorn/longhorn/issues/8656#issue-2320622443).
